pkg/auth/azure.go

```diff
@@ -122,3 +145,80 @@ func (a *Azure) ValidateUser(certificates map[string]CertificateList, token stri
 		Expires: time.Unix(int64(claims["exp"].(float64)), 0),
 	}, nil
 }
+
+func (a *Azure) GroupsForUser(token, email string) (Groups, error) {
+	bearerToken, err := a.getBearerTokenOnBehalfOfUser(token)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodGet, AzureGraphMemberOfEndpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %v", bearerToken))
+	response, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var body []byte
+	response.Body.Read(body)
+
+	var memberOfResponse MemberOfResponse
+	if err := json.NewDecoder(response.Body).Decode(&memberOfResponse); err != nil {
+		return nil, err
+	}
+
+	var groups Groups
+
+	for _, entry := range memberOfResponse.Groups {
+		mail := strings.ToLower(entry.Mail)
+		if !contains("Unified", entry.GroupTypes) || !strings.HasSuffix(mail, "@nav.no") {
```

Comment:
What are we checking for here? What is `Unified`? And shouldn't all of the group e-mails end with `@nav.no`?

Reply:
Dont know what Azure will return, and as a general function to get user groups it is relevant to do the check. If the function is only for this one purpose, we can igore the check.
What do you want?

Reply:
For now it's only for this purpose.